src/background/chrome-api.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * This module provides a wrapper around the Chrome / WebExtension APIs that
+ * the extension uses.
+ *
+ * It has several purposes:
+ *  - Provide a Promise-returning interface for all async APIs
+ *  - Abstract over the differences between different browsers
+ *  - Provide a seam that can be easily mocked in tests
+ */
+
+/**
+ * Wrap the browser APIs exposed via the `chrome` object to return promises.
+ *
+ * This is exposed for testing. Consumers should use {@link chromeAPI}.
+ */
+export function getChromeAPI(chrome = globalThis.chrome) {
+  // In the test environment, the `chrome` global may not exist. This is ignored
+  // for the purposes of determining the return type.
+  if (typeof chrome === 'undefined') {
+    return /** @type {never} */ (null);
+  }
+
+  /**
+   * @template Result
+   * @typedef {(r: Result) => void} Callback
+   */
+
+  /**
+   * Convert an async callback-accepting Chrome API to a Promise-returning version.
+   *
+   * TypeScript may complain if the API has a Manifest V3-only overload that
+   * returns a Promise. Use {@link promisifyAlt} as a workaround.
+   *
+   * @template {any[]} Args
+   * @template Result
+   * @param {(...args: [...Args, Callback<Result>]) => void} fn
+   * @return {(...args: Args) => Promise<Result>}
+   */
+  const promisify =
```

Comment:
This function is copied and tweaked from the `src/background/util.js` module. The main changes are that errors now checked using `chrome.runtime.lastError` instead of the deprecated `chrome.extension.lastError` property, and the addition of a workaround for a typechecking issue with certain APIs that have both callback-accepting and promise-returning versions declared in the `@types/chrome` package.